File: 백준/Silver/11501. 주식/주식.cc
#include <iostream>
#include <stack>
using namespace std;

void solve(){
	int n, x;
	long long ans = 0;
	stack<int> stk;
	cin >> n;
	while(n--){
		cin >> x;
		stk.push(x);
	}
	int mx_cur = stk.top();
	while(!stk.empty()){
		int cur = stk.top(); stk.pop();
		if(cur < mx_cur) ans += (mx_cur-cur);
		else mx_cur = cur;
	}
	cout << ans << '\n';
}

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
	int t;
	cin >> t;
	while(t--) solve();
}

File: 백준/Silver/11501. 주식/주식.java
// Authored by : chjh2129
import java.io.*;
import java.util.*;

public class Main {

  static int n;
  static Stack<Integer> stk = new Stack<>();

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringBuilder sb = new StringBuilder();
    StringTokenizer st;

    int t = Integer.parseInt(br.readLine());
    while(t --> 0){
      // input & init
      n = Integer.parseInt(br.readLine());
      st = new StringTokenizer(br.readLine());
      
      while(st.hasMoreTokens()){
        stk.add(Integer.parseInt(st.nextToken()));
      }

      // solve
      long ans = 0;
      int mx = stk.peek();
      while(!stk.isEmpty()){
        int cur = stk.pop();
        if(cur < mx) ans += mx - cur;
        else mx = cur;
      }

      // StringBuilder mapping
      sb.append(ans).append("\n");
    }

    // output
    System.out.println(sb.toString());
  }
}